core/src/test/scala/com/nvidia/spark/rapids/tool/profiling/AutoTunerSuite.scala

```diff
@@ -1005,4 +1014,117 @@ class AutoTunerSuite extends FunSuite with BeforeAndAfterEach with Logging {
+    val customProps = mutable.LinkedHashMap(
+      "spark.sql.files.maxPartitionBytes" -> "12345678")
+    // mock the properties loaded from eventLog
+    val logEventsProps: mutable.Map[String, String] =
+      mutable.LinkedHashMap[String, String](
+        "spark.executor.cores" -> "16",
+        "spark.executor.instances" -> "1",
+        "spark.executor.memory" -> "80g",
+        "spark.executor.resource.gpu.amount" -> "1",
+        "spark.executor.instances" -> "1",
+        "spark.sql.adaptive.coalescePartitions.minPartitionNum" -> "1",
+        "spark.rapids.shuffle.multiThreaded.reader.threads" -> "8",
+        "spark.rapids.shuffle.multiThreaded.writer.threads" -> "8",
+        "spark.rapids.sql.multiThreadedRead.numThreads" -> "20",
+        "spark.shuffle.manager" -> "com.nvidia.spark.rapids.spark311.RapidsShuffleManager",
+        "spark.sql.shuffle.partitions" -> "1000",
+        "spark.sql.files.maxPartitionBytes" -> "1g",
+        "spark.task.resource.gpu.amount" -> "0.25",
+        "spark.rapids.memory.pinnedPool.size" -> "5g",
+        "spark.rapids.sql.enabled" -> "true",
+        "spark.plugins" -> "com.nvidia.spark.SQLPlugin",
+        "spark.rapids.sql.concurrentGpuTasks" -> "1")
+    val dataprocWorkerInfo = buildWorkerInfoAsString(Some(customProps), Some(32),
+      Some("212992MiB"), Some(5), Some(4), Some("15109MiB"), Some("Tesla T4"))
+    val infoProvider = getMockInfoProvider(3.7449728E7, Seq(0, 0), Seq(0.4, 0.4), logEventsProps,
+      Some(defaultSparkVersion), Seq(), false, 40.0, 200000000000L)
+    val autoTuner: AutoTuner = AutoTuner.buildAutoTunerFromProps(dataprocWorkerInfo, infoProvider)
+    val (properties, comments) = autoTuner.getRecommendedProperties()
+    val autoTunerOutput = Profiler.getAutoTunerResultsAsString(properties, comments)
+    // scalastyle:off line.size.limit
+    val expectedResults =
+      s"""|
+          |Spark Properties:
+          |--conf spark.executor.cores=8
+          |--conf spark.executor.instances=20
+          |--conf spark.executor.memory=16384m
+          |--conf spark.executor.memoryOverhead=5734m
+          |--conf spark.rapids.memory.pinnedPool.size=4096m
+          |--conf spark.rapids.sql.concurrentGpuTasks=2
+          |--conf spark.sql.files.maxPartitionBytes=3669m
+          |--conf spark.task.resource.gpu.amount=0.125
+          |
+          |Comments:
+          |- 'spark.executor.memoryOverhead' must be set if using 'spark.rapids.memory.pinnedPool.size
+          |- 'spark.executor.memoryOverhead' was not set.
+          |- 'spark.sql.adaptive.enabled' should be enabled for better performance.
+          |- Average JVM GC time is very high. Other Garbage Collectors can be used for better performance.
+          |- ${AutoTuner.classPathComments("rapids.jars.missing")}
+          |- ${AutoTuner.classPathComments("rapids.shuffle.jars")}
+          |""".stripMargin
+    // scalastyle:on line.size.limit
+    assert(expectedResults == autoTunerOutput)
+  }
```

Comment:
Should we have another test that will ignore the recommendation if an old version of the plugin is used?

Reply:
Good question.  So the AutoTuner currently doesn't check plugin version for any of its recommendations so in a sense it is assuming the user is on the latest version.  This would apply for MT shuffle settings, specifically.

I'm not sure we need to check/test as we are trying to align our tools versions with the plugin version so that customer using 23.06 plugin version can use 23.06 tools version, etc.  Let me know what you think.

Reply:
Agree that we should do this.  Can I file an issue to add it for both file cache and MT shuffle?  Looks like we'll need to pull out the jar version from the existing function to make it available for other places.

Reply:
Yes, we can file an issue.
I find that the autotuner can benefit from this if we refactor it well.
We can have list of criteria, the applicability of each criteria, map to plugin version, ..etc.
Then any new feature added should be as simple as defining `apply()` of a new criterionClass

Reply:
Issue filed: https://github.com/NVIDIA/spark-rapids-tools/issues/362